ikasaneip/component/endpoint/test-endpoint/Readme.md

```diff
@@ -1,17 +1,118 @@
+[<< Component Quick Start](../../Readme.md)
 ![IKASAN](../../../developer/docs/quickstart-images/Ikasan-title-transparent.png)
 # Event Generating Consumer
 
-Utility consumer for the generation of ad-hoc events for demonstration or test of flows.
+Utility consumer for the generation of ad-hoc events for demonstration or testing of flows.
 
-##### Configuration Options
+<br/>
+
+##### Operation
+Default behaviour is the generation of real-time "Test Message ###" String events to this consumer.
+This behaviour can be overridden by the ```setEndpointEventProvider(EndpointEventProvider endpointEventProvider)``` where you can provide your own event provider to this consumer.
+
+```java
+org.ikasan.spec.event.MessageListener.onMessage(T);
+org.ikasan.spec.event.ExceptionListener.onException(Throwable);
+```
+
+##### Supported Features
+The following Ikasan features are supported by this component.
+
+||| 
+| :----- | :------: | 
+| **Feature**| **Support** | 
+| Managed Lifecycle| Yes | 
+| Component Configuration| No | 
+| Event Resubmission| Yes | 
+| Event Record/Replay| Yes | 
+
+##### Mandatory Configuration Options
+No mandatory configuration required.
+
+##### Optional Configuration Options
 
 | Option | Type | Purpose |
 | --- | --- | --- |
-| identifier | String | Identifier to be set on the generated FlowEvent |
-| payload | String | Payload to be set on the generated FlowEvent |
-| eventGenerationInterval | long | Event generation inrerval in milliseconds.0 = immediate and continuous event generation. This is the default value. |
-| batchSize | int | Used in conjunction with the eventGenerationInterval to determine number of generated events per interval. |
-| eventLimit | String | Allow a limit to be set on the total number of events generated. Default -1 = unlimited. |
+| eventEndpointProvider | EventEndpointProvider | Custom implementation of the EventEndpointProvider contract.|
+
+###### Sample Usage - Ikasan Java FluentAPI
+Example below shows how easy it is to provide your own custom event endpoint provider.
+
+```java
+public class ModuleConfig 
+{
+  @Resource
+  private BuilderFactory builderFactory;
+
+  public Consumer getScheduledConsumer() 
```

Comment:
this method should be getEventGeneratingConsumer

Reply:
getEventGeneratingConsumer